src/components/confirm/confirm.test.tsx

```diff
@@ -0,0 +1,217 @@
+    expect(onCancel).toHaveBeenCalledTimes(1);
+  });
+
+  it("should render disabled confirm button when disableConfirm is set", () => {
+    render(<Confirm open onConfirm={() => {}} disableConfirm />);
+
+    expect(screen.getByRole("button", { name: "Yes" })).toBeDisabled();
+  });
+
+  it("should render disabled cancel button and close icon when disableCancel is set", () => {
+    render(
+      <Confirm
+        open
+        onConfirm={() => {}}
+        onCancel={() => {}}
+        showCloseIcon
+        disableCancel
+      />
+    );
+
+    expect(screen.getByRole("button", { name: "No" })).toBeDisabled();
+    expect(screen.getByRole("button", { name: "Close" })).toBeDisabled();
+  });
+
+  it("should not call onCancel when disableCancel is set and ESC key is pressed", async () => {
+    const onCancel = jest.fn();
+    const user = userEvent.setup();
+    render(
+      <Confirm
+        open
+        onConfirm={() => {}}
+        onCancel={() => {}}
+        showCloseIcon
+        disableCancel
+      />
+    );
+
+    await user.keyboard("{esc}");
+    expect(onCancel).not.toHaveBeenCalled();
+  });
+
+  it("should render disabled confirm button with Loader if isLoadingConfirm is set", () => {
+    render(
+      <Confirm
+        open
+        onConfirm={() => {}}
+        isLoadingConfirm
+        confirmButtonDataProps={{ "data-role": "confirm-button" }}
+      />
+    );
+
+    expect(
+      screen.queryByRole("button", { name: "Yes" })
+    ).not.toBeInTheDocument();
+    expect(screen.getByRole("progressbar", { name: "Loading" })).toBeVisible();
+    expect(screen.getByTestId("confirm-button")).toBeDisabled();
+  });
+
+  it('should render with expected styles when iconType is set to "warning"', () => {
+    render(<Confirm open iconType="warning" onConfirm={() => {}} />);
+
+    const icon = screen.getByTestId("icon");
+    expect(icon).toBeVisible();
+    expect(icon).toHaveStyle({ color: "var(--colorsSemanticCaution500)" });
+  });
+
+  it('should render with expected styles when iconType is set to "error"', () => {
+    render(<Confirm open iconType="error" onConfirm={() => {}} />);
+
+    const icon = screen.getByTestId("icon");
+    expect(icon).toBeVisible();
+    expect(icon).toHaveStyle({ color: "var(--colorsSemanticNegative500)" });
+  });
+
+  it("should render buttons with expected data tags", () => {
+    render(
+      <Confirm
+        open
+        onConfirm={() => {}}
+        onCancel={() => {}}
+        showCloseIcon
+        confirmButtonDataProps={{
+          "data-role": "confirm-button",
+          "data-element": "foo",
+        }}
+        cancelButtonDataProps={{
+          "data-role": "cancel-button",
+          "data-element": "bar",
+        }}
+        closeButtonDataProps={{
+          "data-role": "close-button",
+          "data-element": "baz",
+        }}
+      />
+    );
+
+    expect(screen.queryByTestId("confirm-button")).toHaveAttribute(
```

Comment:
```suggestion
    expect(screen.getByRole("button", {name: "Yes")).toHaveAttribute(
    expect(screen.getByRole("button", {name: "No")).toHaveAttribute(

```
(Non-blocking) I think the recommended approach is to use getByRole wherever possible

Reply:
I went with this approach to check that the button was accessible by the new "data-role" being set, but I can change this to use `getByRole` and then assert the data-role attribute is set also correctly if thats preferred.